server/state.go

```diff
@@ -205,8 +206,19 @@ func (us *userState) getClientState(sessionID string) UserStateClient {
 }
 
 func (cs *callState) getHostID(botID string) string {
-	var host userState
+	if cs.HostLocked {
+		return cs.HostID
+	}
 
+	// if current host is still in the call, keep them as the host
+	for _, state := range cs.Sessions {
```

Comment:
Do we absolutely need to loop twice over sessions?

Reply:
No, but this was done for clarity. I made the change, but I think it's worse for understandability...

Reply:
With the comment I think it's more than reasonable. It's a rather simple rule after all.

Reply:
Ok